crates/provider-sdk/src/lib.rs

```diff
@@ -104,6 +104,17 @@ pub struct Context {
     pub tracing: HashMap<String, String>,
 }
 
+impl Context {
+    /// Get link name from the request.
+    ///
+    /// While link name should in theory *always* be present, it is not natively included in [`Context`] yet,
```

Comment:
Should we have it be `default` if it returns `None` or is that not correct behavior?

Reply:
Here the failure to get the `link-name` header does not mean that there isn't a link name (or that it's `default`), which is why I left it at `None`.

It's unfortunate because we *know* that a link name should be present, but with how it's implemented right now, we can't *ensure* that the value is present, thus we have to look for it (and possibly not find it, no matter what the actual value is).

Reply:
I think the big reason to have it return `default`, potentially with a debug level log when this occurs, is so that newer provider versions can run on 1.0 hosts and still work 🤔 

The other downside there I think is that for all hosts post 1.1 that header should always be there, so it feels like an extra case to handle when it's unlikely.

Bottom line I can't really think of what we would do in our providers other than `link_name.unwrap_or("default")` so I figure we can just go ahead and set that, but if you have a case in mind where we wouldn't handle it that way I think we can also just choose to treat None as default

Reply:
> I think the big reason to have it return default, potentially with a debug level log when this occurs, is so that newer provider versions can run on 1.0 hosts and still work 🤔

I'm not sure I understand -- newer providers would be using a version of the SDK that makes it part of `Context` right -- so it *should* be impossible for them to run into this.

If in the future we properly make it required, then this function can stay and will *always* return `Some(..)`

> The other downside there I think is that for all hosts post 1.1 that header should always be there, so it feels like an extra case to handle when it's unlikely.

Yeah definitely agree on this, but until that header is actually properly part of the contract, it feels like we shouldn't promise it *is* there, or worse give a wrong value... for it to be guaranteed to be there, `Context` has to change, and that requires breaking changes/future versions depending on how hard we semver.

> Bottom line I can't really think of what we would do in our providers other than link_name.unwrap_or("default") so I figure we can just go ahead and set that, but if you have a case in mind where we wouldn't handle it that way I think we can also just choose to treat None as default in our impls

IMO there's a difference between the SDK making the wrong choice for you and leaving you to do it -- you generally don't want the SDK to do the hopeful but possibly wrong thing. 

That said, I'm not against adding this case to the documentation and doing an `unwrap_or`!

Reply:
Added this at the SDK level!